_pages/guides/data-hub-framework.markdown

```diff
@@ -5,17 +5,110 @@ lead_text: ''
+
+The advantages of this approach include: 
+- Combining the Grove-related artifacts with the Data Hub simplifies development and deployments.  The Grove code is logically separated inside its own directory under `data-hub/src/main/ui-modules` for clarity and distinction from Data Hub specific modules.
+- If code is maintained together in the same repository, consistent versioning can be ensured.
+- The Grove UI becomes a downstream consumer of the Data Hub.
+
+There are some downsides to this approach:
+- UI specific code will reside in close proximity to backend specific configurations.  This might not be as preferable to some development teams.
+- Any CRUD operations performed by the Grove project will write to the Data Hub's FINAL database.
+
+
+### How to implement the recommended approach, step-by-step:
+
+Prerequisite: MarkLogic Data Hub has been deployed to your MarkLogic host.  
+
+[Optional] If you previously installed Grove to its default database via the `mlDeploy` command within the `marklogic` directory, remove the Grove-specific configs before proceeding:
+        
+    > cd grove/marklogic
+    > ./gradlew mlUndeploy
+    
+
+### Steps to move Grove modules to the Data Hub
+
+1. Determine the MarkLogic host and port the data-hub-FINAL app server is running on.
+2. At the command line, from within the top level of your Grove project, (the directory with `middle-tier` and `ui` directories inside it), run `grove config` and enter the host and port of the data-hub-FINAL app server.  This will update environment variables for your project.
+3. Create the `ui-modules` directory inside the Data Hub project: `data-hub/src/main/ui-modules`.
+4. Update the Data Hub's `gradle.properties` to register this new location for MarkLogic module deployments.  Add the following line to `gradle.properites`:
+
+    ```
+    mlModulePaths=src/main/ml-modules,src/main/ui-modules
+    ```
+
+5. __Only necessary for Data Hub 5.0.1__ Update the Data Hub's `build.gradle` to use a newer version of ml-gradle.  Add the `dependencies` to the `buildscript` object around line 5: 
+
+    ```JSON
+    dependencies {
+		  classpath "gradle.plugin.com.marklogic:ml-gradle:3.14.0"
+    }
+    ```
+
+6. Copy the contents of the Grove modules to the Data Hub project. Copy 
+    ```
+    grove/marklogic/ml-modules/options
+    grove/marklogic/ml-modules/root 
+    grove/marklogic/ml-modules/services
+    grove/marklogic/ml-modules/transforms
+    
+    to 
+    
+    data-hub/src/main/ui-modules
+    ```
+
+7. Copy the contents of the Grove default user profile and a dictionary document to the Data Hub project. Copy 
+    ```
+    grove/marklogic/ml-content
+    
+    to 
+    
+    data-hub/src/main
+    ```
```

Comment:
ml-content inside src/main is not deployed out of the box by ml-gradle afaik. This might require copying over the appropriate mlcp task from the grove build.gradle.

Reply:
less required anyhow..

Reply:
searched ml-gradle for ml-content, no hits..

Reply:
On second review, there is built-in support for loading files, but it is called mlLoadData, and it looks by default for src/main/ml-data.. I suggest adding an instruction to rename the folder to `ui-data`, and adding the following line to `gradle.properties`: `mlDataPaths=src/main/ml-data,src/main/ui-data`

That seems to do the trick. mlLoadData is part of mlDeploy..

Reply:
Added mlDataPaths and updated ui-data.